torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1390,6 +1390,25 @@ def sample_inputs_addcmul_addcdiv(op_info, device, dtype, requires_grad, **kwarg
 
     return tuple(sample_inputs)
 
+def sample_inputs_attn(op_info, device, dtype, requires_grad, **kwargs):
+    # Three inputs: q, k, v that are all 2-D tensors (m × n).
+    # qₘ = kₘ = vₘ = m
+    # qₙ = kₙ = n
+    # vₙ is unconstrained
+    #
+    # Thus there are three shape variables that can vary
+    # independently: {m, n, vₙ}. The Cartesian cube of an integral set,
+    # sizes here, generates test cases for (m, n, vₙ).
+    sizes = torch.tensor([1, S, M, L])
+    test_cases = torch.cartesian_prod(sizes, sizes, sizes)
```

Comment:
+1 point for using torch functions only !
If you want to do that with python lists, you can use `itertools.product()` as well

Reply:
does the +1 mean that this was a good way to do it? my instinct would be to do it in pure python if i'm not benefiting from pytorch stuff, but is that the wrong instinct?

Reply:
It's a style point :D
But yes I think it would be nicer to read in pure python.

Reply:
done.